### PR TITLE
Switch to GitHub Actions for CI pipelines

### LINKED ISSUE #65: Move CI to Github Actions
Set up CI on Azure Pipelines on #16, but tried Github Actions and it was really nice. Would like to move CI in-repo using GAs.

I should've listened to @mfiumara in the first place. :)

### PR DESCRIPTION
**Pull request by qwandor**
This fixes #65. There are some behaviour differences:

- I'm only using stable Rust, not beta or nightly. I can add them if you want.
- I run Clippy on all there OSes rather than just Linux, and the results get posted back to GitHub as annotations so they show up on the 'Files changed' tab of the PR.
- I've added test coverage, uploaded to [codecov.io](https://app.codecov.io/gh/deviceplug/btleplug).

Build time comparison:

| Job                | Azure  | GitHub Actions |
| ------------------ | ------ | -------------- |
| Linux build/test   | 2m 41s | 2m 52s         |
| MacOS build/test   | 3m 4s  | 5m 38s         |
| Windows build/test | 4m 30s | 3m 45s         |
| Format             | fails  | 55s            |
| Linux coverage     |        | 2m 45s         |
| MacOS coverage     |        | 2m 56s         |
| Windows coverage   |        | 4m 4s          |

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          RUSTFLAGS: "-Zinstrument-coverage"
+          LLVM_PROFILE_FILE: "test-coverage-%p-%m.profraw"
+      - name: Convert coverage
+        run: grcov . -s . --binary-path target/debug/ -t lcov --branch --ignore-not-existing -o target/debug/lcov.info
+      - name: Upload coverage to codecov.io
+        uses: codecov/codecov-action@v1
+        with:
+          directory: ./target/debug
+          fail_ci_if_error: true
+          verbose: true
```
Comment (NZSmartie) [inline .github/workflows/rust.yml:97]: Do we need the verbosity? the output for the code coverage task is very noisy. each line in the script is echoed out, not sure if this is the cause

**Comment by qwandor** [comment]
I'm not sure how to stop the Azure pipelines from being expected, now that they are no longer run. Maybe there's a repository setting somewhere @qdot will need to change?

**Comment by NZSmartie** [comment]
> 
> 
> I'm not sure how to stop the Azure pipelines from being expected, now that they are no longer run. Maybe there's a repository setting somewhere @qdot will need to change?

Yeah, settings are under _Settings_ -> _Branches_ -> _Require status checks to pass before merging_

Example screen shot from one of my own projects
image

**Review by qdot — APPROVED** [review]

**Comment by qdot** [comment]
I removed the branch requirements for azure for now.

**Pull request merged**